erpnext/stock/doctype/material_request/material_request.py

```diff
@@ -159,6 +159,24 @@ def validate(self):
 		self.reset_default_field_value("set_warehouse", "items", "warehouse")
 		self.reset_default_field_value("set_from_warehouse", "items", "from_warehouse")
 
+		self.validate_pp_qty()
+
+	def validate_pp_qty(self):
+		for item in self.items:
+			if item.material_request_plan_item:
+				qty_from_pp = frappe.db.get_value(
+					"Material Request Plan Item",
+					item.material_request_plan_item,
+					["quantity", "requested_qty"],
+					as_dict=1,
```

Comment:
Write a single query outside of the loop to get the data in the dictionary and use that dict here